### ISSUE TITLE
Segment environment sent as "dev-xx-cloud"

### ISSUE THREAD

**Issue by nataliekwong** [airbytehq/airbyte#16006]
# Current Behavior
Segment events are being sent with the prefix "dev-" instead of the previous version, which was the version number, like this: "0.39.27-alpha-cloud"

It looks like this starting happening around July 18th. Not a pressing issue, but I can imagine it may come in handy at some point to have the correct version numbers passing to measure metrics like connector health over versions. 

# Expected Behavior
Instead of "dev", use the version number for Cloud (I'm not sure if "alpha" was deliberately removed or not)

**Comment by octavia-squidington-iii** [comment]
cc @airbytehq/frontend

**Comment by timroes** [comment]
This is intended behavior. We're no longer deploying Cloud from an OSS release as we did beforehand, but nowadays Cloud just pulls in a specific commit of the OSS repository (and which one will be updated when we do a release to the latest). The version number thus we use on cloud is just `dev-<commit hash>-cloud` indicating which commit in the OSS repository we used. There musn't be an equivalent OSS release for this commit, i.e. cloud version numbers are currently not existing in the semantic versioning schema anymore.

cc @letiescanciano for awareness
